Test/DurableTask.AzureStorage.Tests/TestTablePartitionManager.cs

```diff
@@ -0,0 +1,633 @@
+    using System;
+    using System.Diagnostics;
+    using System.Threading.Tasks;
+    using System.Linq;
+
+    [TestClass]
+    public class TestTablePartitionManager
+    {
+
+        string connection = TestHelpers.GetTestStorageAccountConnectionString(); 
+
+        //[TestCategory("DisabledInCI")]
+        [TestMethod]
+        // Start with one worker and four partitions.
+        // Test the worker could claim all the partitions in 5 seconds.
+        public async Task TestOneWorkerWithFourPartition()
+        {
+            //string storageConnectionString = TestHelpers.GetTestStorageAccountConnectionString();
+            //var storageAccount = CloudStorageAccount.Parse(storageConnectionString);
+            string testName = "TestOneWorkerWithFourPartition";
+            var settings = new AzureStorageOrchestrationServiceSettings
+            {
+                TaskHubName = testName,
+                StorageConnectionString = this.connection,
+                WorkerId = "0",
+                AppName = testName,
+                UseTablePartitionManagement = true,
+            };
+
+            TimeSpan timeout = TimeSpan.FromSeconds(5);
+            Stopwatch stopwatch = new Stopwatch();
+            var service = new AzureStorageOrchestrationService(settings);
+            await service.StartAsync();
+
+
+            bool isAllPartitionClaimed = false;
+            stopwatch.Start();
+            while (stopwatch.Elapsed < timeout && !isAllPartitionClaimed)
+            {
+                var partitions = service.ListTableLeases();
+                foreach (var partition in partitions)
+                {
+                    if (partition.CurrentOwner == null) break;
+                    Assert.AreEqual("0", partition.CurrentOwner);
+                    if (partition.RowKey == "controlqueue-03") isAllPartitionClaimed = true;
+                }
+            }
+
+            stopwatch.Stop();
+            await service.StopAsync();
+            await service.DeleteAsync();
+        }
+
+        //[TestCategory("DisabledInCI")]
+        [TestMethod]
+        //Starts with two workers and four partitions.
+        //Test that one worker can acquire two partitions. 
+        //Since two worker can not start at the same time, and one will start earlier than the another one.
+        //There would be a steal process, and test that the lease tranfer will take no longer than 30 sec.
+        public async Task TestTwoWorkerWithFourPartitions()
+        {
+            TimeSpan timeout = TimeSpan.FromSeconds(30);
+            Stopwatch stopwatch = new Stopwatch();
+
+            var services = new AzureStorageOrchestrationService[2];
+            string testName = "TestTwoWorkerWithFourPartitions";
+            var settings = new AzureStorageOrchestrationServiceSettings[2];
+
+            for (int i = 0; i < 2; i++)
+            {
+                settings[i] = new AzureStorageOrchestrationServiceSettings
```

Comment:
It seems like you don't need an array for the settings. Can you instead use a local variable?

```suggestion
                var settings = new AzureStorageOrchestrationServiceSettings
```

This comment applies to all your other tests too.

Reply:
The `AzureStorageOrchestrationServiceSettings` is different via each `AzureStorageOrchestrationService` for the `workerId`. And the `worekrId ` is used to set the `worekrName` for the instance. 

If I used the same one and changed the `workerId` for different services, all the service will then have the same worker Name and thus theywill cause a problem. (They read the table, thought itself had some partitions but actually it might be no. It might be the other worker who shares the same name has the leases.)

This can be solved by adding a new attribute to `TablePartitionManager` class called `workerName`, and pass `workerName` to the `tableLeaseManager` constructor. I didn't do this because in the old versions, partition manager got their names directly by settings.workerId. Seems like there is no need to have a `workerName` attribute considering all the instances should have different workerId normally, and settings is a required attribute.  (We talked a little about this last week. This is basically why I asked haha ) What do you think of it?  I would love to hear your ideas. Thanks!

Reply:
I think you might have misunderstood what I meant. I'm not suggesting they all use the same settings object. I'm simply stating that you don't need to save these settings objects in an array.